docs/development/ceps/proposed/cep-002.rst

```diff
@@ -0,0 +1,194 @@
+.. _cep-002:
+
+
+***********************
+CEP 2 - Event Structure
+***********************
+
+* Status: draft
+* Discussion: NA
+* Date accepted: NA
+* Last revised: 2023-04-04
+* Author: Maximilian Linhoff
+* Created: 2023-04-04
+
+Abstract
+========
+
+Currently, the hierarchy of the ctapipe ``ArrayEventContainer`` container has the data
+levels first and then each data level has a ``Map`` ``.tel`` for telescope specific
+information in subcontainers.
+This CEP proposes to change this structure to have a container ``TelescopeEventContainer``
+as parent for all telescope-wise information that then contains the data levels.
+
+This has the following advantages:
+
+* At lower data levels, many processing steps can be performed independently for each
+  telescope event, which is easier if all data from a single telescope is in a single container.
+* The current proposed ACADA-DPPS ICD fore-sees event and monitoring files per telescope.
+  This maps nicely to readers filling a ``TelescopeEventContainer`` which are then
+  joined together with array level information into an ``ArrayEventContainer``.
+
+Proposed new structure
+======================
+
+This CEP proposes to change the current layout of ``ArrayEventContainer`` from having multiple
+data levels each with a ``Map`` containing telescope-wise data to a structure
+where each ``ArrayEventContainer`` is composed of one to many ``TelescopeEvents`` containing
+all telescope-wise information for all data levels.
+
+The ``ArrayEventContainer`` should also be renamed to ``SubarrayEventContainer``, to match with other naming
+patterns in ctapipe, such as the ``SubarrayDescription``, making it clear that the array is split
+into multiple subarrays, each observing their own observation block.
+
+The main structure after the change will look like this (Container-suffix left out for readability):
+
+.. code-block::
+
+   SubarrayEvent
+   - index: SubarrayEventIndex
+   - trigger: SubarrayTrigger
+   - simulation: SimulatedShower
+   - dl0: DL0Subarray
+   - dl1: DL1Subarray
+   - dl2: DL2Subarray
+   - dl3: DL3Subarray
+   - tel: Map[int, TelescopeEvent] # or just a list? TelescopeEvent stores the tel_id
```

Comment:
This question mark seems like the only remaining issue to decide on. I'm not sure what the idea with the map is, maybe you could describe what options you are envisioning?

Reply:
No, I think also having the map is handy